src/stability_sdk/client.py

```diff
@@ -104,30 +119,147 @@ def process_artifacts_from_answers(
+            )
+        )
+        for resp in self.api._asset.stub.Generate(request, wait_for_ready=True):
+            for artifact in resp.artifacts:
+                if artifact.type == generation.ARTIFACT_TEXT:
+                    return json.loads(artifact.text)
+        raise Exception(f"Failed to load project file for {self.id}")
+
+    def save_settings(self, data: dict) -> str:
+        contents = json.dumps(data)
+        request = generation.Request(
+            engine_id=self.api._asset.engine_id,
+            prompt=[generation.Prompt(
+                artifact=generation.Artifact(
+                    type=generation.ARTIFACT_TEXT,
+                    text=contents,
+                    mime="application/json",
+                    uuid=self.file_id
+                )
+            )],
+            asset=generation.AssetParameters(
+                action=generation.ASSET_PUT, 
+                project_id=self.id, 
+                use=generation.ASSET_USE_PROJECT
+            )
+        )
+        for resp in self.api._asset.stub.Generate(request, wait_for_ready=True):
+            for artifact in resp.artifacts:
+                if artifact.type == generation.ARTIFACT_TEXT:
+                    self.update(file_id=artifact.uuid, file_uri=artifact.text)
+                    logger.info(f"Saved project file {artifact.uuid} for {self.id}")
+                    return artifact.uuid
+        raise Exception(f"Failed to save project file for {self.id}")
+
+    def put_image_asset(
+        self, 
+        image: Union[Image.Image, np.ndarray],
+        use: generation.AssetUse=generation.ASSET_USE_OUTPUT
+    ):
+        prompt = generation.Prompt(
+            parameters=generation.PromptParameters(init=True),
+            artifact=generation.Artifact(
+                type=generation.ARTIFACT_IMAGE,
+                binary=image_to_png_bytes(image)
+            )
+        )
+
+        store_rq = generation.Request(
+            engine_id=self.api._asset.engine_id,
+            prompt=[prompt],
+            asset=generation.AssetParameters(
+                action=generation.ASSET_PUT, 
+                project_id=self.id, 
+                use=use
+            )
+        )
+
+        for resp in self.api._asset.stub.Generate(store_rq, wait_for_ready=True):
+            for artifact in resp.artifacts:
+                if artifact.type == generation.ARTIFACT_TEXT:
+                    return artifact.uuid
+        raise Exception(f"Failed to store image asset for project {self.id}")
+
+    def update(self, title:str=None, file_id:str=None, file_uri:str=None):
+        file = project.ProjectAsset(
+            id=file_id,
+            uri=file_uri,
+            use=project.PROJECT_ASSET_USE_PROJECT,
+        ) if file_id and file_uri else None
+        updated_project = self.api._proj_stub.Update(project.UpdateProjectRequest(
+            id=self.id, 
+            title=title,
+            file=file
+        ))
+        if title:
```

Comment:
following the 'property' pattern I recommended earlier, this update would just need to be
```python
    self._project = updated_project
```
since attributes like 'title' and 'file_id' would evaluate from the hidden _project object directly, no additional internal update logic required.